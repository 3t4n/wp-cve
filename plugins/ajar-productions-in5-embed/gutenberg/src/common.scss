/**
 * #.# Common SCSS
 *
 * Can include things like variables and mixins
 * that are used across the project.
*/

// Colors.
$black: rgb(41, 41, 41);
$white: #f4f4f4;
$gray: #dedede;
$green: #bada55;
$red: orangered;
$dark-gray-800: #23282d;
$light-gray-500: #e2e4e7; // Good for "grayed" items and borders.

// Fonts
$editor-html-font: Menlo, Consolas, monaco, monospace;
$text-editor-font-size: 14px;
