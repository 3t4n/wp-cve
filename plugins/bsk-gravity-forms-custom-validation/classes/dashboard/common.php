<?php
class BSK_GFCV_Dashboard_Common {
	
	public function __construct( $args ) {
		
	}
	
	
    public static function bsk_gfcv_get_form_plugin() {
		global $wpdb;
        
        $entries_table_name = $wpdb->prefix . BSK_GFCV::$_bsk_gfcv_entries_tbl_name;
        $results = $wpdb->get_results( 'SELECT DISTINCT( `forms` ) from `'.$entries_table_name.'` ORDER BY `forms` ASC' );
        if( !$results || !is_array( $results ) || count( $results ) < 1 ){
            return false;
        }
        
        $data_to_return = array();
        foreach( $results as $form_name_obj ){
            $form_name_title = '';
            if ( isset( BSK_GFCV::$_supported_plugins[$form_name_obj->forms] ) ){
                $form_name_title = BSK_GFCV::$_supported_plugins[$form_name_obj->forms]['title'];
            }else{
                switch ( $form_name_obj->forms ) {
                    case 'FF':
                        $form_name_title = 'Formidable Forms';
                    break;
                    case 'GF':
                        $form_name_title = 'Gravity Forms';
                    break;
                    case 'WPF':
                        $form_name_title = 'WPForms';
                    break;
                    case 'NJF':
                        $form_name_title = 'Ninja Forms';
                    break;
                    case 'CF7':
                        $form_name_title = 'Contact Form 7';
                    break;
                }
            }
            $data_to_return[$form_name_obj->forms] = $form_name_title;
        }
        
		return $data_to_return;
	}
	
    public static function bsk_gfcv_get_gf_forms( $form_plugin ) {
		global $wpdb;
        
        $entries_table_name = $wpdb->prefix . BSK_GFCV::$_bsk_gfcv_entries_tbl_name;
        
        $data_to_return = array();
        //Gravity Forms
        if ( $form_plugin == 'GF' && isset( BSK_GFCV::$_supported_plugins['GF'] ) ){
            if ( ! function_exists( 'get_plugins' ) ) {
				require_once ABSPATH . 'wp-admin/includes/plugin.php';
			}
            $gf_table_name = 'rg_form';
            if ( version_compare( BSK_GFCV::$_supported_plugins['GF']['version'], '2.2', '>=' ) ){
                 $gf_table_name = 'gf_form';
            }
            
            $gf_table_name = $wpdb->prefix . $gf_table_name;
            
            $sql = 'SELECT DISTINCT( E.`form_id` ), G.`title` from `'.$entries_table_name.'` AS E '.
                   'LEFT JOIN `'.$gf_table_name.'` AS G ON E.`form_id` = G.`id` '.
                   'WHERE E.`forms` LIKE "GF" ORDER BY G.`title` ASC';
            $results = $wpdb->get_results( $sql );
            if( !$results || !is_array( $results ) || count( $results ) < 1 ){
                return false;
            }
            
            foreach( $results as $gf_form_data ){
                $data_to_return[$gf_form_data->form_id] = $gf_form_data->title;
            }
        }
        
        if ( $form_plugin == 'FF' && isset( BSK_GFCV::$_supported_plugins['FF'] ) ){
            $forms = FrmForm::getAll(
                                        array(
                                            'is_template' => 0,
                                            'status'      => 'published',
                                            array(
                                                'or'               => 1,
                                                'parent_form_id'   => null,
                                                'parent_form_id <' => 1,
                                            ),
                                        ),
                                        'name'
                                    );
            if ( $forms && is_array( $forms ) && count( $forms ) > 0 ) {
                $sql = 'SELECT DISTINCT( E.`form_id` ) from `'.$entries_table_name.'` AS E '.
                       'WHERE E.`forms` LIKE "FF"';
                $results = $wpdb->get_results( $sql );
                if( !$results || !is_array( $results ) || count( $results ) < 1 ){
                    return false;
                }
                $entries_existing_form_ids = array();
                foreach( $results as $ff_form_existing ){
                    $entries_existing_form_ids[] = $ff_form_existing->form_id;
                }
                
                $forms_data_array = array();
                foreach( $forms as $ff_form_data ){
                    if( in_array($ff_form_data->id, $entries_existing_form_ids ) ){
                        $data_to_return[$ff_form_data->id] = $ff_form_data->name;
                    }
                }
            }
        }
        
        if ( $form_plugin == 'WPF' && isset( BSK_GFCV::$_supported_plugins['WPF'] ) ){
            
            $forms = wpforms()->form->get();
            
            if ( $forms && is_array( $forms ) && count( $forms ) > 0 ) {
                $sql = 'SELECT DISTINCT( E.`form_id` ) from `'.$entries_table_name.'` AS E '.
                       'WHERE E.`forms` LIKE "WPF"';
                $results = $wpdb->get_results( $sql );
                if( !$results || !is_array( $results ) || count( $results ) < 1 ){
                    return false;
                }
                $entries_existing_form_ids = array();
                foreach( $results as $wpf_form_existing ){
                    $entries_existing_form_ids[] = $wpf_form_existing->form_id;
                }
                
                $forms_data_array = array();
                foreach( $forms as $wpf_form_data ){
                    if( in_array($wpf_form_data->ID, $entries_existing_form_ids ) ){
                        $data_to_return[$wpf_form_data->ID] = $wpf_form_data->post_title;
                    }
                }
            }
        }

        if ( $form_plugin == 'CF7' && isset( BSK_GFCV::$_supported_plugins['CF7'] ) ){
            $sql = 'SELECT `ID`, `post_title` FROM `'.$wpdb->posts.'` WHERE `post_type` LIKE "wpcf7_contact_form" '.
                   'AND `ID` IN( SELECT DISTINCT( E.`form_id` ) from `'.$entries_table_name.'` AS E '.
                   'WHERE E.`forms` LIKE "CF7" )';
            $results = $wpdb->get_results( $sql );
            if( !$results || !is_array( $results ) || count( $results ) < 1 ){
                return false;
            }

            foreach( $results as $cf7_form_existing ){
                $data_to_return[$cf7_form_existing->ID] = $cf7_form_existing->post_title;
            }
            
        }
        
		return $data_to_return;
	}
    
    public static function bsk_gfcv_render_entry_html( $form_submit_data, $hits_data, $entry_id, $ip, $form_plugin ) {
		global $wpdb;
                
        $hits_tbl = $wpdb->prefix.BSK_GFCV::$_bsk_gfcv_hits_tbl_name;
        $list_tbl = $wpdb->prefix.BSK_GFCV::$_bsk_gfcv_list_tbl_name;
        $items_tbl = $wpdb->prefix.BSK_GFCV::$_bsk_gfcv_items_tbl_name;

        if( $hits_data == false && $entry_id ){
            //get hits data
            $sql = 'SELECT H.*, L.`list_name`, L.`list_type`, L.`check_way`, L.`extra` AS list_extra, I.`value` '.
                   'FROM `'.$hits_tbl.'` AS H LEFT JOIN `'.$list_tbl.'` AS L ON H.list_id = L.id '.
                   'LEFT JOIN `'.$items_tbl.'` AS I ON H.`item_id` = I.id '.
                   'WHERE H.`entry_id` = %d';
            $sql = $wpdb->prepare( $sql, $entry_id );
            $hits_data_results = $wpdb->get_results( $sql );
            if( $hits_data_results && is_array( $hits_data_results ) && count( $hits_data_results ) > 0 ){
                $hits_data = array();
                foreach( $hits_data_results as $hit_data_obj ){
                    if( !isset( $hits_data[$hit_data_obj->field_id] ) ){
                        $hits_data[$hit_data_obj->field_id] = array();
                    }
                    $hits_data[$hit_data_obj->field_id]['list_id'] = $hit_data_obj->list_id;
                    $hits_data[$hit_data_obj->field_id]['list_name'] = $hit_data_obj->list_name;
                    $hits_data[$hit_data_obj->field_id]['list_type'] = $hit_data_obj->list_type;
                    $hits_data[$hit_data_obj->field_id]['list_check_way'] = $hit_data_obj->check_way;
                    $hits_data[$hit_data_obj->field_id]['list_extra'] = $hit_data_obj->list_extra;
                    $hits_data[$hit_data_obj->field_id]['extra_data'] = $hit_data_obj->extra_data;
                    if( !isset( $hits_data[$hit_data_obj->field_id]['items_value'] ) ){
                        $hits_data[$hit_data_obj->field_id]['items_value'] = array();
                    }
                    $hits_data[$hit_data_obj->field_id]['items_value'][] = ( $hit_data_obj->item_id == -1 || $hit_data_obj->item_id == -2 ) ? 'NO_ITEM_MATCH' : $hit_data_obj->value;
                }
            }
        }

        if( !$hits_data || !is_array( $hits_data ) || count( $hits_data ) < 1 ){
            return '<p>No valid keywords hit data</p>';
        }
        
        ob_start();
        
        $form_data = maybe_unserialize( $form_submit_data );
        if( $form_data && is_array( $form_data ) && count( $form_data ) > 0 ){
        ?>
        <div class="bsk-gfcv-entry-form-data-container">
            <table class="widefat striped">
                <thead>
                    <?php if ( $form_plugin == 'CF7' ) { ?>
                    <th>Field Name</th>
                    <?php } else { ?>
                    <th>Field ID</th>
                    <th>Field label</th>
                    <?php } ?>
                    <th>Field value</th>
                    <th>&nbsp;</th>
                </thead>
                <tbody>

        <?php
        foreach( $form_data as $field_ID => $field_data ){
            if( $field_ID == 'form_id' ){
                continue;
            }
            $blocked_info = '';
            if( $hits_data && isset( $hits_data[$field_ID] ) ){
                $blocked_data = $hits_data[$field_ID];
                $blocked_item_extra_data = maybe_unserialize( $hits_data[$field_ID]['extra_data'] );

                $blocked_info .= 'blocked by list: ';


                $blocked_list_id = intval( $hits_data[$field_ID]['list_id'] );
                $list_edit_url = admin_url( 'admin.php?page='.BSK_GFCV_Dashboard::$_bsk_gfcv_pages['base']['slug'].'&view=edit&id='.$blocked_list_id );

                $blocked_info .= '<a href="'.$list_edit_url.'">'.$hits_data[$field_ID]['list_name'].'</a>';
                $blocked_info .= '<span class="bsk-gfcv-entry-blocked-keyword" style="display: block;">'.$blocked_item_extra_data['CV_message'].'</span>';
            }
        ?>
                <tr>
                    <?php if ( $form_plugin == 'CF7' ) { ?>
                    <td class="bsk-gfcv-column-ID"><?php echo $field_ID; ?></td>
                    <?php } else { ?>
                    <td class="bsk-gfcv-column-ID"><?php echo $field_ID; ?></td>
                    <td class="bsk-gfcv-column-label"><label><?php echo $field_data['label']; ?></label></td>
                    <?php } ?>
                    <td class="bsk-gfcv-column-value"><?php echo $field_data['value']; ?></td>
                    <td class="bsk-gfcv-column-blocked-info"><?php echo $blocked_info; ?></td>
                </tr>
        <?php
        }
        ?>
                </tbody>
            </table>
        </div>
        <?php
        }
        
        $entry_html = ob_get_contents();
        ob_end_clean();
        
        return $entry_html;
	}
    
    public static function bsk_gfcv_get_mail_tmpl(){
        require_once( 'common-tmpl.php' );
        
        return $email_html_tmpl;
    }

    public static function bsk_gfcv_is_form_plugin_supported( $form_plugin ){
        
        //check license type

        $settings_data = get_option( BSK_GFCV_Dashboard::$_plugin_settings_option, false );
        $supported_form_plugins = array( 'GF' );
        if( $settings_data && is_array( $settings_data ) && count( $settings_data ) > 0 ){
            if( isset( $settings_data['supported_form_plugins'] ) && count( $settings_data['supported_form_plugins'] ) > 0 ){
                $supported_form_plugins = $settings_data['supported_form_plugins'];
            }
        }

        return ( in_array( $form_plugin, $supported_form_plugins ) && isset( BSK_GFCV::$_supported_plugins[$form_plugin] ) );
    }

    public static function bsk_gfcv_get_list_by_type( $list_type, $selected ){
		global $wpdb;
		
		if( $list_type == "" ){
			return '';
		}
		
        $list_table = $wpdb->prefix.BSK_GFCV::$_bsk_gfcv_list_tbl_name;
		$options_str = '';
		
		$sql = 'SELECT * FROM `'.$list_table.'` WHERE `list_type` = %s ORDER BY `list_name` ASC';
		$sql = $wpdb->prepare( $sql, $list_type );
		$results = $wpdb->get_results( $sql );
		if( $results && is_array($results) && count($results) > 0 ){
			foreach( $results as $list_obj ){
				$checked_str = $list_obj->id == $selected ? ' selected' : '';
				$options_str .= '<option value="'.$list_obj->id.'"'.$checked_str.'>'.$list_obj->list_name.'</option>';
			}
		}
		
		return $options_str;
	}

    public static function get_ip() {

		$ip = $_SERVER['REMOTE_ADDR'];

		// HTTP_X_FORWARDED_FOR can return a comma separated list of IPs; use the first one
		$ips = explode( ',', $ip );

		return $ips[0];
	}

}
