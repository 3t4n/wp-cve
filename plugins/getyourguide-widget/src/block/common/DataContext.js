import { createContext } from 'react';

export default createContext({
  currency: {
    USD: {
      value: 'USD',
      label: 'US$ US Dollar',
    },
    EUR: {
      value: 'EUR',
      label: '€ Euro',
    },
    AUD: {
      value: 'AUD',
      label: 'AU$ Australian Dollar',
    },
    GBP: {
      value: 'GBP',
      label: '￡ British Pound',
    },
    CAD: {
      value: 'CAD',
      label: 'C$ Canadian Dollar',
    },
    CSK: {
      value: 'CSK',
      label: 'Kč Czech koruna',
    },
    DKK: {
      value: 'DKK',
      label: 'DKK Danish Krone',
    },
    HKD: {
      value: 'HKD',
      label: 'HK$ Hong Kong Dollar',
    },
    JPY: {
      value: 'JPY',
      label: '¥ Japanese Yen',
    },
    NZD: {
      value: 'NZD',
      label: 'NZ$ New Zealand Dollar',
    },
    NOK: {
      value: 'NOK',
      label: 'NOK Norwegian krone',
    },
    PLN: {
      value: 'PLN',
      label: 'zł Polish Złoty',
    },
    SGD: {
      value: 'SGD',
      label: 'S$ Singapore Dollar',
    },
    SEK: {
      value: 'SEK',
      label: 'SEK Swedish Krona',
    },
    CHF: {
      value: 'CHF',
      label: 'CHF Swiss Franc',
    },
    AED: {
      value: 'AED',
      label: 'د.إ UAE Dirham',
    },
  },
  locale_code: {
    'da-DK': { value: 'da-DK', label: 'Dansk' },
    'de-DE': { value: 'de-DE', label: 'Deutsch' },
    'de-AT': { value: 'de-AT', label: 'Deutsch (Österreich)' },
    'de-CH': { value: 'de-CH', label: 'Deutsch (Schweiz)' },
    'en-US': { value: 'en-US', label: 'English' },
    'en-GB': { value: 'en-GB', label: 'English (United Kingdom)' },
    'es-ES': { value: 'es-ES', label: 'Español' },
    'es-MX': { value: 'es-MX', label: 'Español (México)' },
    'fr-FR': { value: 'fr-FR', label: 'Français' },
    'it-IT': { value: 'it-IT', label: 'Italiano' },
    'nl-NL': { value: 'nl-NL', label: 'Nederlands' },
    'no-NO': { value: 'no-NO', label: 'Norsk' },
    'pl-PL': { value: 'pl-PL', label: 'Polski' },
    'pt-PT': { value: 'pt-PT', label: 'Português' },
    'pt-BR': { value: 'pt-BR', label: 'Português (Brasil)' },
    'fi-FI': { value: 'fi-FI', label: 'Suomen kieli' },
    'sv-SE': { value: 'sv-SE', label: 'Svenska' },
    'tr-TR': { value: 'tr-TR', label: 'Türkçe' },
    'ru-RU': { value: 'ru-RU', label: 'Русский' },
    'ja-JP': { value: 'ja-JP', label: '日本語' },
    'zh-CN': { value: 'zh-CN', label: '简体中文' },
    'zh-TW': { value: 'zh-TW', label: '繁體中文' },
  },
});
