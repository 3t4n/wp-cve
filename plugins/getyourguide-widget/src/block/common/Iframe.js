// @flow
import React from 'react';
import { stringify } from 'query-string';
import throttle from 'lodash.throttle';
import {
  activitiesDefaultAttributes, cityDefaultAttributes, gygData, spacers,
} from '../config';

const iframeMsgListener = (target) => {
  const el = target;
  if (!el) {
    return;
  }
  window.addEventListener(
    'message',
    (msg) => {
      if (
        msg.origin !== 'https://widget.getyourguide.com'
        || parseInt(el.style.height, 10) === msg.data.height
        || msg.data.url !== el.firstChild.src
      ) {
        return;
      }
      el.style.height = `${msg.data.height + 1}px`;
    },
    false,
  );
};

const postMessage = (target) => {
  if (!target) {
    return;
  }
  target.firstChild.contentWindow.postMessage('update height', 'https://widget.getyourguide.com');
};

const handleIframeResize = (target) => {
  window.addEventListener('resize', throttle(postMessage.bind(null, target), 1000));
};

const iframeHandlers = (target) => {
  if (!target) {
    return;
  }
  iframeMsgListener(target);
  postMessage(target);
  handleIframeResize(target);
};

const qs = ({ widgetType = 'activities', queries = {} }) => {
  const { partnerID: partner_id } = gygData; // eslint-disable-line camelcase
  const allQueries = (() => {
    switch (widgetType) {
    case 'city':
      return { ...cityDefaultAttributes, ...queries, partner_id };
    default:
      return { ...activitiesDefaultAttributes, ...queries, partner_id };
    }
  })();
  return stringify(
    Object.keys(allQueries).reduce(
      (acc, key) => (allQueries[key] && allQueries[key].length > 0 ? { [key]: allQueries[key], ...acc } : acc),
      {},
    ),
  );
};
const title = widgetType => `GetYourGuide ${widgetType} widget`;
const url = 'https://widget.getyourguide.com/default/';

type IframeProps = {
  queries?: Object,
  widgetType?: string,
  wpWidgetType?: string,
};
class Iframe extends React.Component<IframeProps> {
  container = React.createRef();

  static defaultProps = {
    queries: {},
    widgetType: 'activities',
    wpWidgetType: 'wp_activities',
  };

  componentDidMount() {
    const { current } = this.container;
    iframeHandlers(current);
  }

  render() {
    const { container } = this;
    const { widgetType = 'activities', wpWidgetType = 'wp_activities', queries } = this.props;
    return (
      <div ref={container}>
        <iframe
          src={`${url}${widgetType}.frame?widget=${wpWidgetType}&${qs({
            widgetType,
            queries,
          })}`}
          style={{ height: '100%', marginBottom: spacers.standard }}
          title={title(widgetType)}
        />
      </div>
    );
  }
}

const wpIframe = ({ widgetType, wpWidgetType, queries }) => {
  const { createElement } = window.wp.element;
  return createElement(
    'div',
    {
      class: 'gyg-iframe-container',
      style: {
        marginBottom: spacers.standard,
      },
    },
    createElement('iframe', {
      style: {
        border: '0',
        height: '100%',
        width: '100%',
      },
      title: title(widgetType),
      src: `${url}${widgetType}.frame?widget=${wpWidgetType}&${qs({
        widgetType,
        queries,
      })}`,
    }),
  );
};

export { Iframe, wpIframe, iframeHandlers };
