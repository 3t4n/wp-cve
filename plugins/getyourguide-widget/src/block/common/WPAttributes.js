// @flow
import { Component } from 'react';

type WPAttributesProps = {
  attributes: Object,
  children: React.Node,
  defaultAttributes: Object,
  setAttributes: Function,
};

export default class WordPress extends Component<WPAttributesProps> {
  constructor(props) {
    super(props);
    const { defaultAttributes } = this.props;
    this.labels = Object.keys(defaultAttributes).reduce((acc, key) => {
      switch (key) {
      case 'cmp':
        return { ...acc, cmp: "Track your widget's performance with a campaign name (optional)" };
      case 'iata':
        return { ...acc, iata: 'IATA Airport Code' };
      case 'currency':
        return { ...acc, currency: 'Currency' };
      case 'locale_code':
        return { ...acc, locale_code: 'Language' };
      case 'number_of_items':
        return { ...acc, number_of_items: 'Number of items' };
      case 'q':
        return { ...acc, q: 'Search query' };
      default:
        return acc;
      }
    }, {});
    this.state = defaultAttributes;
  }

  componentDidUpdate(_, prevState) {
    const { attributes, defaultAttributes, setAttributes } = this.props;
    setAttributes(
      Object.keys({ ...defaultAttributes, ...attributes }).reduce((acc, key) => {
        const { [key]: update } = this.state;
        return prevState[key] !== update ? { ...acc, [key]: update } : acc;
      }, {}),
    );
  }

  handleChange = (key = '', e = { preventDefault: () => null, target: { value: '' } }) => {
    e.preventDefault();
    this.setState({ [key]: e.target.value });
  };

  render() {
    const { labels, handleChange, state } = this;
    const { attributes, children, defaultAttributes } = this.props;
    return children({
      labels,
      state,
      attributes,
      defaultAttributes,
      handleChange,
    });
  }
}
