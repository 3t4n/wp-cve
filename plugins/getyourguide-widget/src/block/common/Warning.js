import React from 'react';

const Warning = () => (
  <strong className="notice notice-error" style={{ display: 'block' }}>
    {"You haven't entered your GetYourGuide partner ID."}
    {' '}
    <a href={`${window.location.origin}/wp-admin/options-general.php?page=getyourguide`}>
      Enter it here.
    </a>
  </strong>
);

export default Warning;
