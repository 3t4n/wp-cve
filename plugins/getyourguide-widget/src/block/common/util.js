// @flow
import React, { Fragment } from 'react';

export const Input = ({
  label,
  value,
  onChange,
  ...passThrough
}: {
  label: string,
  value: string,
  onChange: Function,
}) => (
  <div>
    <label>
      {label}
      <input
        {...{
          style: { display: 'block' },
          onChange,
          value,
          ...passThrough,
        }}
      />
    </label>
  </div>
);

export const Select = ({
  data = {},
  value,
  onChange,
  label,
}: {
  data: Object,
  value: string,
  onChange: Function,
  label: string,
}) => (
  <label style={{ display: 'block' }}>
    {label}
    <select style={{ display: 'block' }} value={value} onBlur={onChange} onChange={onChange}>
      {Object.keys(data).map((option) => {
        const { value: optionValue, label: optionLabel } = data[option];
        return (
          <option key={optionLabel} {...{ value: optionValue }}>
            {optionLabel}
          </option>
        );
      })}
    </select>
  </label>
);

export const Inputs = ({
  data,
  keys,
  labels,
  onChange,
  selects = [],
  values,
}: {
  data: Object,
  keys: Array<string>,
  labels: Object,
  onChange: Function,
  selects: Array<string>,
  values: Object,
}) => (
  <Fragment>
    {keys.map((key) => {
      if (selects.includes(key)) {
        return (
          <Select
            key={key}
            {...{
              data: data[key],
              label: labels[key],
              value: values[key],
              onChange: e => onChange(key, e),
            }}
          />
        );
      }
      return (
        <Input
          key={key}
          {...{
            label: labels[key],
            onChange: e => onChange(key, e),
            type: Number.isInteger(parseInt(values[key], 10)) ? 'number' : 'text',
            value: values[key],
          }}
        />
      );
    })}
  </Fragment>
);
