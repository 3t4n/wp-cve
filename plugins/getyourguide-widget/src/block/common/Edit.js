import React, { Component } from 'react';
import DataContext from './DataContext';
import { Iframe } from './Iframe';
import Warning from './Warning';
import { gygData } from '../config';

class Edit extends Component {
  state = {
    optionsDisplay: 'flex',
  };

  render() {
    const {
      children,
      defaultAttributes,
      attributes,
      labels,
      handleChange,
      state,
      widgetType,
    } = this.props;
    const { optionsDisplay: display } = this.state;
    const keys = Object.keys({ ...defaultAttributes, ...attributes });
    return (
      <DataContext.Consumer>
        {data => (
          <div>
            {gygData.partnerID.length === 0 && <Warning />}
            <button
              className="components-button"
              type="button"
              style={{ float: 'right' }}
              onClick={() => this.setState(({ optionsDisplay }) => ({
                optionsDisplay: optionsDisplay === 'flex' ? 'none' : 'flex',
              }))
              }
            >
              {`${display === 'none' ? 'Show' : 'Hide'} options`}
            </button>
            <div
              className="components-placeholder"
              style={{
                alignItems: 'initial',
                display,
                marginBottom: '10px',
                textAlign: 'left',
              }}
            >
              {children({
                data,
                keys,
                defaultAttributes,
                attributes,
                labels,
                handleChange,
                state,
              })}
            </div>
            <Iframe {...{ widgetType, queries: { ...defaultAttributes, ...attributes } }} />
          </div>
        )}
      </DataContext.Consumer>
    );
  }
}

export default Edit;
