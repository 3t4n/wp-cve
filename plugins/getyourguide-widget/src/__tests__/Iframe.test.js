import React from 'react';
import renderer from 'react-test-renderer';
import { Iframe } from '../block/common/Iframe';

describe('Iframe', () => {
  it('renders correctly with no props', () => {
    const tree = renderer.create(<Iframe />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('renders a city widget', () => {
    const tree = renderer.create(<Iframe widgetType="city" />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
