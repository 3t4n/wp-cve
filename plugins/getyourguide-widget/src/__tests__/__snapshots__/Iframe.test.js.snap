// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Iframe renders a city widget 1`] = `
<div>
  <iframe
    src="https://widget.getyourguide.com/default/city.frame?widget=city&iata=SYD&locale_code=en-US"
    style={
      Object {
        "height": "100%",
        "marginBottom": "1.5em",
      }
    }
    title="GetYourGuide city widget"
  />
</div>
`;

exports[`Iframe renders correctly with no props 1`] = `
<div>
  <iframe
    src="https://widget.getyourguide.com/default/activities.frame?widget=activities&currency=USD&locale_code=en-US&number_of_items=3"
    style={
      Object {
        "height": "100%",
        "marginBottom": "1.5em",
      }
    }
    title="GetYourGuide activities widget"
  />
</div>
`;
